packages/runtime/container-runtime/src/opLifecycle/remoteMessageProcessor.ts

```diff
@@ -24,11 +24,27 @@ export class RemoteMessageProcessor {
 
     public process(remoteMessage: ISequencedDocumentMessage): ISequencedDocumentMessage {
         let message = copy(remoteMessage);
-
-        message = this.opDecompressor.processMessage(message);
+        message = this.opDecompressor.processMessage(message).message;
         unpackRuntimeMessage(message);
-        message = this.opSplitter.processRemoteMessage(message);
 
+        const chunkProcessingResult = this.opSplitter.processRemoteMessage(message);
+        message = chunkProcessingResult.message;
+        if (chunkProcessingResult.state !== "Processed") {
```

Comment:
I have not thought about all pros/cons of this design, but let me describe alternative:

With the current design only runtime can uncompress content (because it's chunked after compression). The ideal solution seems like the one where we keep compression content (ops) as they come in, and the moment compressed size is about to exceed 1Mb, we flush buffer by sending one compressed op, that is a chunk (I.e. it can't be processed individually). 
The two benefits of this approach are:
1. remote clients can start processing (decompressing) content sooner, as it comes in.
2. service can recompress ops for long-term storage (like uncompressing them, and running compression for while file)
3. attached ops have better treatment (that's basically same as # 1, with an extra benefit of being able to apply results to data model, as we can "close" a batch when flushing, i.e. have many independent batches).

This will make generating pipeline more complex, but I think receiving (this code) part of pipeline will be simpler, as
1. Compression is independent from anything else. If op is compressed, it gets uncompressed and any trace of compression is removed, the rest of pipeline should not care if there was compression
2. After that, it's either a chunk (and chunking code takes care of it), OR it's a new "group op" (an op that has many attach ops, but single sequence number). Long terms I think we can get rid of the later (via virtualizing sequence numbers).